### DISCUSSION TITLE
[Bug]: preset-env with useBuiltIns: 'usage' fails with Node < 18 if no DOMException constructor polyfill

### DISCUSSION THREAD

**Discussion by tkrotoff** [babel/babel#15264] [General]
If you have a dependency that uses DOMException (in my case https://github.com/getsentry/sentry-javascript/blob/v3.27.2/packages/utils/src/supports.ts#L48)

- With `useBuiltIns: 'usage'`, `webpack --mode=production` fails with Node < 18 if no DOMException constructor polyfill (web.dom-exception.constructor):

```
ERROR in ./src/index.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
HookWebpackError: Cannot read properties of undefined (reading 'prototype')
    at tryRunOrWebpackError (node_modules/webpack/lib/HookWebpackError.js:88:9)
    [...]
-- inner error --
TypeError: Cannot read properties of undefined (reading 'prototype')
    at Object.<anonymous> (node_modules/core-js/modules/web.dom-exception.stack.js:33:74)
```

https://github.com/zloirock/core-js/blob/v3.26.1/packages/core-js/modules/web.dom-exception.stack.js#L33

- With `useBuiltIns: 'entry'`, `webpack --mode=production` works, I don't understand why the difference

Part of the cause is that Node < 18 doesn't have DOMException: https://developer.mozilla.org/en-US/docs/Web/API/DOMException#browser_compatibility

- Solution is to force the inclusion of web.dom-exception.constructor:

```JS
// babel.config.js
module.exports = {
  sourceType: 'unambiguous',

  presets: [
    [
      '@babel/preset-env',
      {
        useBuiltIns: 'usage',

        // THIS IS THE FIX
        include: ['web.dom-exception.constructor'],

        corejs: '3.26'
      }
    ],
    ['@babel/preset-react', { runtime: 'automatic' }],
    '@babel/preset-typescript'
  ]
};
```

**Comment by nicolo-ribaudo** [comment]
Do you have a repository I could use to reproduce the bug? `web.dom-exception.stack` and `web.dom-exception.constructor` have exactly the same compat data for Node.js (both supported starting from `17.0`), so I don't see how Babel could load `web.dom-exception.stack` without also loading `web.dom-exception.constructor` :thinking:
